packages/base-manager/src/manager-base.ts

```diff
@@ -327,12 +326,14 @@ export abstract class ManagerBase implements IWidgetManager {
     );
   }
 
-  register_model(model_id: string, modelPromise: Promise<WidgetModel>): void {
+  async register_model(
+    model_id: string,
+    modelPromise: Promise<WidgetModel>
+  ): Promise<void> {
```

Comment:
Coulnd't this change the overall code, 
previously this wasn't async, now the caller might be tempted into awating on this when they didn't in the past,
and such an await could have other ripple effects, which could potentially lead to race conditions or the like.

For instance if `modelPromise` failed in the past, then the call to `reigster_model` would'nt fail, but now it would.

Reply:
The call to `model.once` could be wrapped in a try catch block to prevent `register_model` from failing. 

As far as the ripple effects of converting a previously sync method to async - is that the nature of converting something to async? Maybe there is a way to signal that this async method doesn't need to be awaited? I can look around to see if this is a pain point with async-conversions. I'm new to async stuff so this is an interesting consideration to me.

Reply:
Maybe I'm misunderstanding something with chaining promises, but a rejected promise forces the program execution to jump to nearest rejection handler. In the past case, there was no `.catch()` chained to `modelPromise`, therefore the call to `register_model` would also fail, wouldn't it? 

Regardless, reverting this function to non-async. Because it doesn't make sense to `await` here since it would essentially just be blocking the thread and definitely can cause unnecessary race conditions.